core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -488,9 +488,16 @@ private[spark] class ExecutorAllocationManager(
     newExecutorTotal = numExistingExecutors
     if (testing || executorsRemoved.nonEmpty) {
       executorsRemoved.foreach { removedExecutorId =>
+        // If it is cachedBlcok timeout is configured using
+        // spark.dynamicAllocation.cachedExecutorIdleTimeout
+        val idleTimeout = if (blockManagerMaster.hasCachedBlocks(removedExecutorId)) {
```

Comment:
How about changing `removeTimes` to `HashMap[String, (Long, Boolean)]` (and the `Boolean` field indicates whether it is for cachedExecutor idle timeout or not) ? Thus, we do not need to ask `blockManagerMaster` again.

Reply:
blockManagerMaster already provides API to check it is cached block or not,so I feel it will be overhead to maintain another HashMap

Reply:
We do not maintain another HashMap, but alter its original structure.  In this way,  we do not need to issue extra rpc calls to `BlockManagerMaster` here. As you mentioned 'API', this thing happens after a rpc call happened.